node-core/components/interfaces.go

```diff
@@ -48,7 +48,6 @@ type (
 	AttributesFactory interface {
 		BuildPayloadAttributes(
 			st *statedb.StateDB,
-			slot math.Slot,
```

Comment:
slot is always picked from state (or from the block after ProcessSlots has been called, which makes state slots equal to block state).
So we can drop this parameter